testing/regress/hpcc/regression/cleanup.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+import re
+import requests
+import json
+import time
+import glob
+from datetime import datetime
+
+# Configure logger
+logger = logging.getLogger('cleanup') 
+
+# Custom logger formatter for serial numbering of log records
+class SerialNumberFormatter(logging.Formatter):
+    def __init__(self):
+        super().__init__()
+        self.serialNumber = 0
+
+    def format(self, record):
+        self.serialNumber += 1
+        record.msg = str(self.serialNumber) + ". " + record.msg
+        return super().format(record)
+
+# Builds and configures the cleanup logger
+def buildCleanupLogger(logDir, cleanupLogger):
+    cleanupLogger.setLevel(logging.INFO)
+    curTime = time.strftime('%y-%m-%d-%H-%M-%S')
+    logName = "cleanup" + "." + curTime + ".log"
+    logPath = os.path.join(logDir, logName)
+    cleanupHandler = logging.FileHandler(logPath)
+    cleanupHandler.setFormatter(SerialNumberFormatter())
+    cleanupLogger.addHandler(cleanupHandler)
+    return cleanupLogger
+
+# Loads configuration data from JSON file
+def loadConfig(filename):
+    with open(filename, 'r') as file:
+        return json.load(file)
+
+# Extracts regress log files corresponding to current run of RTE
+def getRegressLogs(mode, logDirPath, startTime, endTime):
+    regressLogFiles = glob.glob(os.path.join(logDirPath, '*.log'))   
+    for logFilePath in regressLogFiles:
```

Comment:
This solution is working.
However, try to thinking about a developer/tester, like me who has >5000 result log files from the lasts 3~4 years in my 'logDirPath'. Do you really need to check all file names in that directory? You have start and end time. Please, thinking about it.